packages/components/tabs/src/TabsList.tsx

```diff
@@ -0,0 +1,183 @@
+      if (!('current' in listRef) || !listRef.current) {
+        return
+      }
+
+      if (orientation !== 'horizontal') {
+        setWithArrows({ prev: 'hidden', next: 'hidden' })
+      } else {
+        setWithArrows({
+          prev: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+          next: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+        })
+      }
+    }, [orientation, listRef, width])
+
+    useEffect(() => {
+      /**
+       * Enable/disable arrows
+       */
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current || withArrows.prev === 'hidden' || loop) {
+        return
+      }
+
+      const toggleArrowsVisibility = (target: HTMLDivElement) => {
+        setWithArrows({
+          prev: target.scrollLeft > 0 ? 'visible' : 'disabled',
+          next:
+            target.scrollLeft + target.clientWidth < target.scrollWidth ? 'visible' : 'disabled',
+        })
+      }
+
+      const currentList = listRef.current
+
+      toggleArrowsVisibility(currentList)
+
+      currentList.addEventListener('scroll', ({ target }) =>
+        toggleArrowsVisibility(target as HTMLDivElement)
+      )
+
+      return () =>
+        currentList.removeEventListener('scroll', ({ target }) =>
+          toggleArrowsVisibility(target as HTMLDivElement)
+        )
+    }, [listRef, withArrows.prev, loop])
+
+    const handlePrevClick = useCallback(() => {
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current) {
+        return
+      }
+
+      const shouldLoopForward = loop && listRef.current.scrollLeft <= 0
+
+      listRef.current.scrollTo({
+        left: shouldLoopForward
+          ? listRef.current.scrollLeft + listRef.current.scrollWidth - listRef.current.clientWidth
+          : listRef.current.scrollLeft - listRef.current.clientWidth,
+        behavior: 'smooth',
+      })
+    }, [listRef, loop])
+
+    const handleNextClick = useCallback(() => {
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current) {
+        return
+      }
+
+      const shouldLoopBackward =
+        loop &&
+        listRef.current.scrollLeft + listRef.current.clientWidth >= listRef.current.scrollWidth
+
+      listRef.current.scrollTo({
+        left: shouldLoopBackward ? 0 : listRef.current.scrollLeft + listRef.current.clientWidth,
+        behavior: 'smooth',
+      })
+    }, [listRef, loop])
+
+    return (
+      <div className={wrapperStyles({ className })} ref={wrapperRef}>
+        {withArrows.prev !== 'hidden' && (
+          <Button
+            shape="square"
+            intent="surface"
+            size="sm"
+            className={navigationArrowStyles()}
+            onClick={handlePrevClick}
+            disabled={withArrows.prev === 'disabled'}
+            aria-label="Scroll left"
+          >
+            <Icon size="sm">
```

Comment:
`size` should not be needed here I think 🤔

Reply:
Yes here it isn't, but seemed clearer to me, just as we declared above, for `Button`, a `size="sm"`. Also, what if default size value for icon change? :thinking: